integrations/opensearch/src/haystack_integrations/document_stores/opensearch/document_store.py

```diff
@@ -326,6 +330,11 @@ def _bm25_retrieval(
         if filters:
             body["query"]["bool"]["filter"] = normalize_filters(filters)
 
+        # For some applications not returning the embedding can save a lot of bandwidth
+        # if you don't need this data not retrieving it can be a good idea
+        if not self._return_embedding:
+            body["_source"] = {"excludes": ["embedding"]}
```

Comment:
Curious: Does it make sense to return the embedding at all for keyword retrieval?

Reply:
I know a pipeline in production that does bm25 retrieval (I think) and then uses the returned document vectors to issue an other search via embedding retrieval. With this, the application does not require to compute embeddings and therefore no GPU since the embedding already exist for the first retrieved doc. 

I think we can keep this. Not sure how valid this is.